packages/api/src/hooks/p2p/useOrderList.ts

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import useInfiniteQuery from '../../useInfiniteQuery';
+import useAuthorize from '../useAuthorize';
+
+/** This custom hook returns a list of orders under the current client. */
+const useOrderList = (
+    payload?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_order_list'>>[1]>['payload'],
+    config?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_order_list'>>[1]>['options']
+) => {
+    const { isSuccess } = useAuthorize();
+    const { data, fetchNextPage, ...rest } = useInfiniteQuery('p2p_order_list', {
+        payload: { ...payload, offset: payload?.offset || 0, limit: payload?.limit || 50 },
```

Comment:
```suggestion
        payload: { ...payload, offset: payload?.offset, limit: payload?.limit },
```

Reply:
nit

Reply:
shouldnt we default these to 0 or 50? I remember I had this conversation with jim as he mentions its better to have a fallback value here since payload is optional and possibly undefined

<img width="714" alt="Screenshot 2023-12-12 at 3 33 07 PM" src="https://github.com/binary-com/deriv-app/assets/103412909/8ccbadc4-c103-4ba3-8051-4536f4a0e98f">

Reply:
The default value for limit and offset handled in `useInfiniteQuery` already 👍